### ISSUE TITLE
Please help testing next React-PDF release!

### ISSUE THREAD

**Issue by wojtekmaj** [wojtekmaj/react-pdf#748]
Folks,

React-PDF 5.3.0 is just around the corner, with several exciting updates:

* Updated pdf.js from 2.5.207 to 2.6.347 (#746). Thanks, @andi-dev!
    > This release features support for form filling (AcroForm) and layers (optional content).
* Replaced worker-loader with file-loader in Webpack-specific entry file (#756).
* Replaced PDFLinkService with custom LinkService, which should resolve issues caused by us importing PDFLinkService and EventBus from PDF.js utils designed for browsers only.
    This should resolve https://github.com/wojtekmaj/react-pdf/issues/657 and https://github.com/wojtekmaj/react-pdf/issues/659.
* Slightly reduced bundle size

Please help me with testing [v5.3.0-beta.2](https://github.com/wojtekmaj/react-pdf/releases/tag/v5.3.0-beta.2) by:
* Checking if the issue described in https://github.com/wojtekmaj/react-pdf/issues/657 is gone if this affected you
* Checking if the issue described in https://github.com/wojtekmaj/react-pdf/issues/659 is gone if this affected you
* Checking if internal links (e.g. Table of Contents) work as they used to
* Checking for any additional issues that you may notice
* Checking if entry.webpack still works for you, if you were using Webpack
* Checking if entry.webpack starts working for you, if you were using CRA with external CDN previously

Please comment what you checked and whether your test was successful.

Your help is invaluable. Thank you in advance!

**Comment by imustafin** [comment]
This is about #659.

I am using `react-rails`. My previous comment https://github.com/wojtekmaj/react-pdf/issues/659#issuecomment-707038492

I upgraded react-pdf to 5.3.0-beta:
package.json
```
...
    "react-pdf": "^5.3.0-beta",
...
```

yarn.lock
```
...
react-pdf@^5.3.0-beta:
  version "5.3.0-beta"
  resolved "https://registry.yarnpkg.com/react-pdf/-/react-pdf-5.3.0-beta.tgz#a421b75e476b65e18a2b4da8087c4cc1617d94e3"
  integrity sha512-9pRckeUcCOFe+0iNnaSbiltckRdUa6LHr8e5waC77+/+6j2Yz4UAF/9w7ED9RL68EnWxv21GFmFE4DHt5zmvAw==
  dependencies:
    "@babel/runtime" "^7.0.0"
    make-cancellable-promise "^1.0.0"
    make-event-props "^1.1.0"
    merge-class-names "^1.1.1"
    merge-refs "^1.0.0"
    pdfjs-dist "2.6.347"
    prop-types "^15.6.2"
    worker-loader "^3.0.0"

pdfjs-dist@2.6.347:
  version "2.6.347"
  resolved "https://registry.yarnpkg.com/pdfjs-dist/-/pdfjs-dist-2.6.347.tgz#f257ed66e83be900cd0fd28524a2187fb9e25cd5"
  integrity sha512-QC+h7hG2su9v/nU1wEI3SnpPIrqJODL7GTDFvR74ANKGq1AFJW16PH8VWnhpiTi9YcLSFV9xLeWSgq+ckHLdVQ==
...
```

Importing in TS like this:
```
import { Document } from 'react-pdf';
import { Page } from 'react-pdf';
```
or like this:
```
import { Document } from 'react-pdf/dist/esm/entry.webpack';
import { Page } from 'react-pdf/dist/esm/entry.webpack';
```
Still doesn't work.

The error is
```
ERROR in ./node_modules/react-pdf/dist/esm/Page/AnnotationLayer.js 195:8-29
"export 'AnnotationLayer' (imported as 'pdfjs') was not found in 'pdfjs-dist'
    at HarmonyImportSpecifierDependency._getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:109:11)
    at HarmonyImportSpecifierDependency.getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:68:16)
    at Compilation.reportDependencyErrorsAndWarnings (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1463:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1258:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/var/www/app_name/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1253:28)
    at /var/www/app_name/node_modules/webpack/lib/Compiler.js:672:17
    at _done (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at eval (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1185:12
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1097:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11)

ERROR in ./node_modules/react-pdf/dist/esm/entry.webpack.js 14:2-27
"export 'GlobalWorkerOptions' (imported as 'pdfjs') was not found in 'pdfjs-dist'
    at HarmonyImportSpecifierDependency._getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:109:11)
    at HarmonyImportSpecifierDependency.getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:68:16)
    at Compilation.reportDependencyErrorsAndWarnings (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1463:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1258:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/var/www/app_name/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1253:28)
    at /var/www/app_name/node_modules/webpack/lib/Compiler.js:672:17
    at _done (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at eval (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1185:12
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1097:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11)

ERROR in ./node_modules/react-pdf/dist/esm/Document.js 96:28-55
"export 'PDFDataRangeTransport' (imported as 'pdfjs') was not found in 'pdfjs-dist'
    at HarmonyImportSpecifierDependency._getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:109:11)
    at HarmonyImportSpecifierDependency.getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:68:16)
    at Compilation.reportDependencyErrorsAndWarnings (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1463:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1258:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/var/www/app_name/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1253:28)
    at /var/www/app_name/node_modules/webpack/lib/Compiler.js:672:17
    at _done (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at eval (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1185:12
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1097:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11)

ERROR in ./node_modules/react-pdf/dist/esm/Page/PageSVG.js 91:25-42
"export 'SVGGraphics' (imported as 'pdfjs') was not found in 'pdfjs-dist'
    at HarmonyImportSpecifierDependency._getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:109:11)
    at HarmonyImportSpecifierDependency.getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:68:16)
    at Compilation.reportDependencyErrorsAndWarnings (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1463:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1258:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/var/www/app_name/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1253:28)
    at /var/www/app_name/node_modules/webpack/lib/Compiler.js:672:17
    at _done (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at eval (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1185:12
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1097:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11)

ERROR in ./node_modules/react-pdf/dist/esm/Document.js 195:34-51
"export 'getDocument' (imported as 'pdfjs') was not found in 'pdfjs-dist'
    at HarmonyImportSpecifierDependency._getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:109:11)
    at HarmonyImportSpecifierDependency.getErrors (/var/www/app_name/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:68:16)
    at Compilation.reportDependencyErrorsAndWarnings (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1463:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1258:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (/var/www/app_name/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.finish (/var/www/app_name/node_modules/webpack/lib/Compilation.js:1253:28)
    at /var/www/app_name/node_modules/webpack/lib/Compiler.js:672:17
    at _done (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at eval (eval at create (/var/www/app_name/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:20:22)
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1185:12
    at /var/www/app_name/node_modules/webpack/lib/Compilation.js:1097:9
    at processTicksAndRejections (internal/process/task_queues.js:75:11)

```

This fix https://github.com/wojtekmaj/react-pdf/issues/659#issuecomment-720351158 worked in react-pdf 5.2.0.

This also helps in react-pdf 5.3.0-beta

**Comment by paescuj** [comment]
Regarding #657 - it's working for me with Next.js now 👍 Thank you!

**Comment by LuisPerez94** [comment]
Regarding #657 - it's working for me with Next.js now 👍 Thank you!

**Comment by paescuj** [comment]
I get the following warning now:
```
Warning: Failed prop type: Attempted to check if a variable is a File on a non-browser environment.
    at Document ([...]/node_modules/react-pdf/dist/umd/Document.js:78:37)
```

Using Next.js and loading `react-pdf` as follows:
```
import { Document, Page, pdfjs } from 'react-pdf';

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;
```

Otherwise `5.3.0-beta` works great!

**Comment by wojtekmaj** [comment]
@paescuj - Thanks! Fixed this "failed prop type" thing in 8ec1c3a7a56cbf45ff33d7c147b01a48a8404f22.

**Comment by paescuj** [comment]
@wojtekmaj Great, thanks!

**Comment by wojtekmaj** [comment]
Tested 5.3.0-beta.2 with CRA and Webpack 5, works fine with default Webpack entry file.

**Comment by po5i** [comment]
@wojtekmaj I couldn't find any updated instructions at the v5.3.0-beta.2 README file. How does the worker line CDN change for CRA?

**Comment by wojtekmaj** [comment]
@po5i I _just updated the docs: https://github.com/wojtekmaj/react-pdf#create-react-app

tl;dr use Webpack entry or copy `pdf.worker.js` from `node_modules/pdfjs-dist/build` to `/public` and keep using "standard" entry.

**Comment by adderpositive** [comment]
It works fine on my project running on Next.js. I am using 5.3.0-beta.3. Thanks.

**Comment by sbodi10** [comment]
Also had the same issue listed here: https://github.com/wojtekmaj/react-pdf/issues/613#issue-664399914
Updating version to v5.3.0-beta.2 worked!

Thanks @wojtekmaj

**Comment by mahyaa21** [comment]
My project is running on Next.js. I am using 5.3.0-beta.3 version but my document does not load without any errors. It just shows a message that says Loading PDF…. 
Could you please help me with this problem? @wojtekmaj

**Comment by paescuj** [comment]
> My project is running on Next.js. I am using 5.3.0-beta.3 version but my document does not load without any errors. It just shows a message that says Loading PDF….
> Could you please help me with this problem? @wojtekmaj

Have you configured it as described under https://github.com/wojtekmaj/react-pdf#standard-browserify-and-others? Any errors in the console?

**Comment by mahyaa21** [comment]
Thank you so much! My problem is resolved :) @paescuj

**Comment by mahyaa21** [comment]
> > My project is running on Next.js. I am using 5.3.0-beta.3 version but my document does not load without any errors. It just shows a message that says Loading PDF….
> > Could you please help me with this problem? @wojtekmaj
> 
> Have you configured it as described under https://github.com/wojtekmaj/react-pdf#standard-browserify-and-others? Any errors in the console?

Thank you so much! My problem is resolved :)

**Comment by jplew** [comment]
FYI, I was getting this error with `5.2.0`:

```bash
./node_modules/worker-loader/dist/cjs.js!./node_modules/react-pdf/dist/esm/pdf.worker.entry.js
TypeError: Cannot read property 'version' of undefined

> Build error occurred
Error: > Build failed because of webpack errors
    at /home/jplew/Sites/greenflower/lxp/node_modules/next/dist/build/index.js:17:924
    at async Span.traceAsyncFn (/home/jplew/Sites/greenflower/lxp/node_modules/next/dist/telemetry/trace/trace.js:6:584)
```

using this code:

```
import { Document, Page } from 'react-pdf/dist/esm/entry.webpack';
```

Ever since updating to `5.3.0-beta.3` the problem went away. Thanks !

Framework: Next.js 10
Webpack 5
npm@7.0.15
node@v15.4

**Comment by wojtekmaj** [comment]
I see that in general nothing horrible has happened with the beta, so I released v5.3.0 officially. Thanks for your feedback!

**Comment by paescuj** [comment]
Great! Thank you @wojtekmaj!
